redash/models/__init__.py

```diff
@@ -821,20 +821,40 @@ def evaluate(self):
     def subscribers(self):
         return User.query.join(AlertSubscription).filter(AlertSubscription.alert == self)
 
-    def render_template(self):
-        if not self.template:
+    def render_template(self, template):
+        if not template:
             return ''
+
         data = json_loads(self.query_rel.latest_query_data.data)
-        context = {'rows': data['rows'], 'cols': data['columns'], 'state': self.state}
-        return mustache_render(self.template, context)
+        host = base_url(self.query_rel.org)
+
+        col_name = self.options['column']
+        if data['rows'] and col_name in data['rows'][0]:
+            result_value = data['rows'][0][col_name]
+        else:
+            result_value = ''
+
+        context = {
+            'ALERT_NAME': self.name,
+            'ALERT_URL': '{host}/alerts/{alert_id}'.format(host=host, alert_id=self.id),
+            'ALERT_STATUS': self.state.upper(),
+            'ALERT_CONDITION': self.options['op'],
+            'ALERT_THRESHOLD': self.options['value'],
+            'QUERY_NAME': self.query_rel.name,
+            'QUERY_URL': '{host}/queries/{query_id}'.format(host=host, query_id=self.query_rel.id),
+            'QUERY_RESULT_VALUE': result_value,
+            'QUERY_RESULT_ROWS': data['rows'],
+            'QUERY_RESULT_COLS': data['columns'],
+        }
+        return mustache_render(template, context)
 
     @property
-    def template(self):
-        return self.options.get('template', '')
+    def custom_body(self):
+        return self.render_template(self.options['custom_body'])
```

Comment:
Could it be an issue in terms of backward compatibility (replacing `template` option with `custom_body` - and for subject as well - what if somebody already has alerts with custom )?

Also - shouldn't it be `self.options.get('custom_body', '')` as before?

Reply:
> Could it be an issue in terms of backward compatibility

There are several backward incompatible changes in here (like the template variables). Because it's hidden behind a feature flag, I doubt many people use it. 

But to be friendlier to them I suggest we keep the option name as `template`? As for the variables, we will document the change in the release notes.

Reply:
> Could it be an issue in terms of backward compatibility

This feature was off since it landed, so I'm assuming that chances are slim that someone changed `clientConfig.extendedAlertOptions` apart for the original author @k-tomoyasu.

Maybe I can do this?:
```py
# backwards compatibility
def template(self):
  return self.custom_body;
```

> Also - shouldn't it be self.options.get('custom_body', '') as before?

`render_template()` now returns `''` if `custom_body` us empty. Don't think it's enough?

Reply:
`custom_body` feels a better name for this option than `template` - so if you think it's not an issue - I'd prefer new names as well.

About template variables: do we introduce them (or some new ones), or rename/remove some existing?

Reply:
> About template variables: do we introduce them (or some new ones), or rename/remove some existing?

I renamed the original 3:
https://github.com/getredash/redash/blob/cbfd25f9d7cc7c159170f11b81b984d663641f1b/client/app/services/alert-template.js#L6-L10

Reply:
> `render_template()` now returns `''` if `custom_body` us empty. Don't think it's enough?

I think @kravets-levko was referring to the case when `self.options` doesn't have the `custom_body` key (every existing alert). I address this in my example below. If you access `self.options['custom_body']` (or `custom_subject`) elsewhere in the code, you need to add similar treatment.

> Maybe I can do this?:

This won't help. What might happen is that a user will have an existing alert object with the `template` key in `options`, while you look for `custom_body`. To make it backward compatible you can change the `custom_body` property method into something like:

```python
 def custom_body(self):
        template = self.options.get('custom_body', self.options.get('template'))
        return self.render_template(template)
```

This will look in the `custom_body` key but if not found will give a try at looking in `template`. If not found it will return `None`, so you need to update `render_template` as well (to return empty if it's `None`).

Btw, shouldn't the configuration key be `custom_body_template`?

Reply:
@arikfr does this suffice? https://github.com/getredash/redash/pull/4170/commits/5f8634acc92d73cb5d35b8c38e9154c4d17a8f0e